lib/internal/streams/operators.js

```diff
@@ -32,6 +36,27 @@ const {
 const kEmpty = Symbol('kEmpty');
 const kEof = Symbol('kEof');
 
+function compose(stream, options) {
+  if (options != null) {
+    validateObject(options, 'options');
+  }
+  if (options?.signal != null) {
+    validateAbortSignal(options.signal, 'options.signal');
+  }
+
+  const composedStream = staticCompose(this, stream);
```

Comment:
what if signal already aborted, this is already handled in the `addAbortSignalNoValidate` but should we handle it here before static compose

Reply:
The current impl seems fine? What's the benefit of doing it before?

Reply:
thought that maybe avoid creating new stream if we get a function but, it's better like now